tests/bluetooth/iso/modules/central.c

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <stdlib.h>
+#include <ctype.h>
+#include <getopt.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <bluetooth/services/nus.h>
+#include <bluetooth/services/nus_client.h>
+#include <bluetooth/gatt_dm.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(central, CONFIG_ISO_TEST_LOG_LEVEL);
+
+#define REMOTE_DEVICE_NAME_PEER	    CONFIG_BT_DEVICE_NAME
+#define REMOTE_DEVICE_NAME_PEER_LEN (sizeof(REMOTE_DEVICE_NAME_PEER) - 1)
+
+static bool running;
+struct peer_device {
+	struct bt_conn *conn;
+	struct k_work_delayable dummy_data_send_work;
+};
+static struct peer_device remote_peer[CONFIG_BT_MAX_CONN];
+static struct bt_nus_client *nus[CONFIG_BT_MAX_CONN];
+static struct bt_nus_client nus_client[CONFIG_BT_MAX_CONN];
+static struct bt_le_conn_param conn_param = {
+	.interval_min = 32, /* Set default ACL interval min to 40 ms */
+	.interval_max = 32, /* Set default ACL interval max to 40 ms */
+	.latency = 0,	    /* Set default slave latency to 0 */
+	.timeout = 40	    /* Set default supervision timeout to 4000 ms*/
+};
+
+static void start_scan(void);
+static bool all_peers_connected(void);
+
+static int channel_index_get(const struct bt_conn *conn, uint8_t *index)
+{
+	if (conn == NULL) {
+		LOG_ERR("No connection provided");
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(remote_peer); i++) {
+		if (remote_peer[i].conn == conn) {
+			*index = i;
+			return 0;
+		}
+	}
+
+	LOG_WRN("Connection not found");
+
+	return -EINVAL;
+}
+
+static void work_dummy_data_send(struct k_work *work)
+{
+	int ret;
+	char dummy_string[CONFIG_BT_MAX_CONN][sizeof(uint32_t)] = {0};
+	static uint8_t channel_index;
+	static uint32_t acl_send_count[CONFIG_BT_MAX_CONN];
+
+	struct peer_device *peer = CONTAINER_OF(work, struct peer_device, dummy_data_send_work);
+
+	ret = channel_index_get(peer->conn, &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+		return;
+	}
+	acl_send_count[channel_index]++;
+	sys_put_le32(acl_send_count[channel_index], dummy_string[channel_index]);
+
+	ret = bt_nus_client_send(&nus_client[channel_index], dummy_string[channel_index],
+				 sizeof(dummy_string[0]));
+	if (ret) {
+		LOG_DBG("failed to send, ret = %d", ret);
```

Comment:
Might not be error, since this might occur sometimes because of ACL has hard retransmission mechanism, which keeps retransmit a packet until timeout. So we could see this warning pop up if a packet cannot be sent to peer in a connection interval and we try to send a new packet.